File: app/src/main/java/com/oriente/aptsample/MyApplication.java
package com.oriente.aptsample;

import android.app.Application;

import com.oriente.aptsample.manager.StartupManager;
import com.oriente.aptsample.task.Task1;
import com.oriente.aptsample.task.Task2;
import com.oriente.aptsample.task.Task3;
import com.oriente.aptsample.task.Task4;
import com.oriente.aptsample.task.Task5;

public class MyApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        new StartupManager.Builder().addStartup(new Task5()).addStartup(new Task4()).addStartup(new Task3()).addStartup(new Task2()).addStartup(new Task1()).build(this).start();

    }
}

File: app/src/main/java/com/oriente/aptsample/MyApplication.kt
package com.oriente.aptsample

import android.app.Application
import android.os.Debug
import dagger.hilt.android.HiltAndroidApp
import java.text.DateFormat
import java.text.SimpleDateFormat
import java.util.*

@HiltAndroidApp
class MyApplication : Application() {

    override fun onCreate() {
        super.onCreate()

        val dateFormat: DateFormat = SimpleDateFormat("dd_MM_yyyy_hh_mm_ss", Locale.getDefault())
        val logDate: String = dateFormat.format(Date())
        // Applies the date and time to the name of the trace log.
        Debug.startMethodTracing("sample-$logDate")
    }
}